src/OrchardCore.Modules/Orchard.Tokens.Content/ContentTokens.cs

```diff
@@ -8,15 +9,15 @@ namespace Orchard.Tokens
 {
     public static class ContentTokens
     {
-        public static void RegisterContentTokens(this IHandlebars handlebars)
+        public static void RegisterContentTokens(this IHandlebars handlebars, IHttpContextAccessor httpContextAccessor)
         {
             // Renders a slug for the current content item. Do not use to represent the url
             // as the content item as it might be different than the computed slug.
             handlebars.RegisterHelper("slug", (output, context, arguments) =>
             {
-                IServiceProvider serviceProvider = context.ServiceProvider;
-                var contentManager = serviceProvider.GetRequiredService<IContentManager>();
-                var slugService = serviceProvider.GetRequiredService<ISlugService>();
+                var services = httpContextAccessor.HttpContext.RequestServices;
```

Comment:
Hmmm...  does any of this have access to htp context? (output, context, arguments)